src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -36,6 +34,8 @@
 @Component
 public class DatasetRequestValidator implements Validator {
 
+    private final static ImmutableSet VALID_DATATYPES = ImmutableSet.of("fileref", "int64", "float64", "string", "date");
```

Comment:
I think it would be better to avoid using third party libraries if they can be replaced with equivalent JDK constructs. Once we update to JDK 11, this can be done using `Set.of()`. In the meantime, you could use `Collections.unmodifiableSet(new HashSet<>(...))`. Although, since this is a private field and is not visible outside the class, it is fine to not provide this extra level of security.